compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -280,6 +280,11 @@ fn resolve_associated_item<'tcx>(
                 return Ok(None);
             }
 
+            // If the item does not have a value, then we cannot return an instance.
```

Comment:
The above note says:

```rust
// NOTE: This should be kept in sync with the similar code in
// `rustc_trait_selection::traits::project::assemble_candidates_from_impls()`.
```

but in this case, I think it's justified to keep these "out of sync". During typeck, we want to act like an item is there even if the impl is missing, so we don't get weird method probe and associated item resolution regressions. It's just that when we finally want to look for the `Instance` of that item, that should fail, since a concrete instance for it doesn't exist.

Reply:
this actually is consistent with project afaict

https://github.com/rust-lang/rust/blob/96c2df810b0b681fee63cae11ca63844792b6190/compiler/rustc_trait_selection/src/traits/project.rs#L1991-L2001

Reply:
oh, that's great to find haha